src/_pytest/fixtures.py

```diff
@@ -534,6 +534,14 @@ def getfixturevalue(self, argname: str) -> Any:
             f'The fixture value for "{argname}" is not available.  '
             "This can happen when the fixture has already been torn down."
         )
+
+        if (
+            isinstance(fixturedef, FixtureDef)
+            and fixturedef is not None
+            and fixturedef.use_cache is False
+        ):
+            self._fixture_defs.pop(argname)
```

Comment:
as mentioned in the issue before, i think its important to use a scope name here as indicator instead of the new boolean (in particular as use_cache as implemented here invalidates the scope

the scope name we had in mind back in 2020 was "invocation" to mean that the scope of a validity for a fixture value would be the call (be it other fixture or test function in which it was used 

i like the hack with just popping the values as a workaround for the current tech debt as it seems like the impact is limited, but i'd like the input of the others in this

Reply:
I see.
If we insist on going with the scope like approach rather than the boolean, I think we could still work-around this while leaving the implementation the same. We just change the arg name to a string scope and will currently only support one scope for this (or two considering default behavior).
This way changing the implementation to a more robust infrastructure centered around the actual scoping won't be breaking.
I think this would be a step in the right direction and would probably solve the general use-case for this issue :)
Thoughts?

Reply:
That's the rough idea - add a new scope name, then use the workaround you created and gate on the scope name

For good measure we might need a test that ensures setup/teardown is in sync with the usage sites

Reply:
I've made the rough changes as I understand the request.
I have added a new scope `Invocation` which will be a scope lower than `Function` (of course it's cleanup scope will occur during `Function` cleanup level)
Of course if this is the direction we will be going with I'll move the tests to `test_scope.py`
Please take a look and tell me what you think :)